### PR TITLE
add get_build_metadata function

### PR DESCRIPTION
**Pull request by jnewb1**
reads from local `build.json` if available, otherwise falls back to creating manually from the git commands like we talked about @adeebshihadeh

### CODE REVIEW

File: system/version.py [hunk 1]
```diff
@@ -75,6 +80,42 @@ def is_dirty() -> bool:
...
+      "git_commit": commit
+    }
+  }
+
+
+def get_build_metadata(path: str = BASEDIR) -> BuildMetadata | None:
+  build_metadata_path = pathlib.Path(path) / BUILD_METADATA_FILENAME
+
+  if build_metadata_path.exists():
+    return cast(BuildMetadata, json.loads(build_metadata_path.read_text()))
```
[the remark this answers is not in the document]
Comment (jnewb1) [reply system/version.py:109]: no. I can look for a way to enforce the schema

[the remark this answers is not in the document]
Comment (adeebshihadeh) [reply system/version.py:109]: I'd just parse it out like `json_data.get('channel', 'unkown')`

File: system/version.py [hunk 2]
```diff
@@ -75,6 +80,49 @@ def is_dirty() -> bool:
   return dirty
 
 
+class OpenpilotMetadata(TypedDict):
+  version: str
+  release_notes: str
+  git_commit: str
+
+
+class BuildMetadata(TypedDict):
```
Comment (adeebshihadeh) [inline system/version.py:89]: why is this a typed dict at all? why not just a normal dataclass?

Comment (jnewb1) [reply system/version.py:89]: i thought it made sense to leave it as a dict as thats how it's stored on disk as json, but I can make it a dataclass

**Comment by adeebshihadeh** [comment]
Let's merge this then update all the `get_version` type of functions?

**Pull request merged**